src/DistributedTracingManager.ts

```diff
@@ -0,0 +1,251 @@
+
+                if (samplingRate === undefined) {
+                    return;
+                }
+            }
+        }
+
+        TelemetryClient.sendEvent(Constants.IoTHubAIUpdateDistributedSettingStartEvent);
+
+        await vscode.window.withProgress({
+            title: `Update Distributed Tracing Setting`,
+            location: vscode.ProgressLocation.Notification,
+        }, async () => {
+            try {
+                const result = await this.scheduleTwinUpdate(mode, samplingRate, iotHubConnectionString, deviceIds);
+                TelemetryClient.sendEvent(Constants.IoTHubAIUpdateDistributedSettingDoneEvent, { Result: "Success" }, iotHubConnectionString);
+                this.outputLine(Constants.IoTHubDistributedTracingSettingLabel,
+                    `Update distributed tracing setting for device [${deviceIds.join(",")}] complete!` + result);
+            } catch (err) {
+                TelemetryClient.sendEvent(Constants.IoTHubAIUpdateDistributedSettingDoneEvent, { Result: "Fail", Message: err.message }, iotHubConnectionString);
+                this.outputLine(Constants.IoTHubDistributedTracingSettingLabel, `Failed to get or update distributed setting: ${err.message}`);
+                return;
+            }
+        });
+    }
+
+    private async scheduleTwinUpdate(enable: boolean, samplingRate: number, iotHubConnectionString: string, deviceIds: string[]) {
+        let twinPatch = {
+            etag: "*",
+            properties: {
+                desired: {},
+            },
+        };
+
+        if (enable === undefined && samplingRate === undefined) {
+            return;
+        }
+
+        if (!twinPatch.properties.desired[Constants.DISTRIBUTED_TWIN_NAME]) {
+            twinPatch.properties.desired[Constants.DISTRIBUTED_TWIN_NAME] = {};
+        }
+
+        if (enable !== undefined) {
+            twinPatch.properties.desired[Constants.DISTRIBUTED_TWIN_NAME].sampling_mode = enable ? 2 : 1;
+        }
+
+        if (samplingRate !== undefined) {
+            twinPatch.properties.desired[Constants.DISTRIBUTED_TWIN_NAME].sampling_rate = samplingRate;
+        }
+
+        if (deviceIds.length === 1) {
+            return new Promise((resolve, reject) => {
+                let registry = iothub.Registry.fromConnectionString(iotHubConnectionString);
+
+                registry.updateTwin(deviceIds[0], JSON.stringify(twinPatch), twinPatch.etag, (err) => {
+                    if (err) {
+                        return reject(err.message);
+                    } else {
+                        return resolve("");
+                    }
+                });
+            });
+        }
+
+        return await this.updateDeviceTwinJob(twinPatch, iotHubConnectionString, deviceIds);
+    }
+
+    private updateDeviceTwinJob(twinPatch, iotHubConnectionString: string, deviceIds: string[]) {
```

Comment:
This function to better naming `scheduleTwinUpdate`

Reply:
fixed